sphinx/ext/napoleon/docstring.py

```diff
@@ -1067,12 +1060,11 @@ def _consume_field(self, parse_type: bool = True, prefer_type: bool = False
             _name, _type = line, ''
         _name, _type = _name.strip(), _type.strip()
         _name = self._escape_args_and_kwargs(_name)
-        if self._config.napoleon_use_param:
-            _type = _convert_numpy_type_spec(
-                _type,
-                location=self._get_location(),
-                translations=self._config.napoleon_type_aliases or {},
-            )
+        _type = _convert_numpy_type_spec(
+            _type,
+            location=self._get_location(),
+            translations=self._config.napoleon_type_aliases or {},
+        )
```

Comment:
my (naive) understanding before actually working on this was that `napoleon_use_param` was a cryptic way of saying "I want to link parameter types" (and the same for `napoleon_use_rtype`). That doesn't seem to be true, but maybe we should allow opting out of the preprocessor? We could decide based on `napoleon_type_aliases` (don't preprocess if `napoleon_type_aliases is None`), or we can create yet another setting.

Reply:
I don't think the additional configuration is not needed. I don't think `napoleon_use_param = False` means "not to convert types to hyperlinks".

Do you mean that it is better to add a new configuration to control the hyperlink-conversion regardless of `napoleon_use_param`?  Or it controls the conversion only if `napoleon_use_param = False`?  If you meant latter one, I feel it is not needed.

Reply:
both, actually, but since `napoleon_use_param = True` will always link (are we even able to disable that?) it becomes the latter. So if you think this is fine as is, let's merge?

Reply:
Are you saying there is who wants not to create hyperlinks? I don't still understand your concern. I don't object to add a new option. But I need to understand why it is needed. Additionally, it would be better to be orthogonal to `napoleon_use_param` configuration if possible because `napoleon_use_param` does not mean "not to create hyperlinks".

Reply:
> it would be better to be orthogonal to napoleon_use_param configuration

that's what I was thinking, too, but I wouldn't know how

> Are you saying there is who wants not to create hyperlinks?

exactly, but maybe that's something that can be thought about once there are bug reports / feature requests about that?

Reply:
>exactly, but maybe it's better to only think about that once there are bug reports / feature requests?

Yes. I think so.